File: Moncho_MATLAB/R6.m
function im_filtro = R6(im_rule_6)

[fil, col, ch] = size(im_rule_6);
im_YCbCr = rgb2ycbcr(im_rule_6);
im_filtro = double(zeros(fil, col));
th = 70;

% Inicializacion del vector de la imagen filtrada
im_Y = double(zeros(fil, col));
im_Cb = double(zeros(fil, col));
im_Cr = double(zeros(fil, col));

im_Y = double(im_YCbCr(:, :, 1));
im_Cb = double(im_YCbCr(:, :, 2));
im_Cr = double(im_YCbCr(:, :, 3));

for x = 1:fil
    for y = 1:col
        if abs(im_Cb(x, y) - im_Cr(x, y)) >= th
            im_filtro(x, y) = 1;
        else
            im_filtro(x, y) = 0;
        end
    end
end

imshow(im_filtro)
end 

File: Rules_Py/R6.py
import numpy as np
import matplotlib.pyplot as plt
import cv2

def R6(image):
    """Filtro con Regla 6 para deteccion de fuego """
    im_read = cv2.imread(image)

    fil, col, ch = im_read.shape

    im_YCbCr = cv2.cvtColor(im_read, cv2.COLOR_RGB2YCrCb)
    im_filtro = np.zeros((fil, col))
    im_Y = np.zeros((fil, col), dtype=np.double)
    im_Cb = np.zeros((fil, col), dtype=np.double)
    im_Cr = np.zeros((fil, col), dtype=np.double)

    im_Y, im_Cb, im_Cr = cv2.split(im_YCbCr)

    threshold = 70

    for idx in range(0, fil):
        for idy in range(0, col):
            test = np.double(im_Cb[idx][idy]) - np.double(im_Cr[idx][idy])
            if abs(test) >= threshold:
                im_filtro[idx][idy] = 1
            else:
                im_filtro[idx][idy] = 0

    cv2.imshow("Rule 6", im_filtro)
    cv2.waitKey(0)
    cv2.destroyAllWindows()